Packages/com.createneptune.sdk/Runtime/CNExtensions.cs

```diff
@@ -206,15 +205,46 @@ public static void CreateObjectPool(List<GameObject> gameObjectList, GameObject
             }
         }
 
-        public static T GetPooledObject<T>(List<GameObject> objectPool, GameObject gameObject,
-            int physicsLayer, Transform t, Vector3 positionOffset, Quaternion rotationOffset, bool ignoreParentTransform)
+        /// <summary>
+        /// Returns an object from the specified pool. Creates a new object and adds it to the pool if necessary.
+        /// </summary>
+        /// <param name="objectPool">The active pool</param>
+        /// <param name="gameObject">The prefab to be used to create a new object</param>
+        /// <param name="parent">The transform under which a new object should be placed</param>
+        /// <returns>An active object from the requested pool</returns>
+        private static GameObject GetPooledObject(List<GameObject> objectPool, GameObject gameObject, Transform parent)
         {
-            return GetPooledObject(objectPool, gameObject, physicsLayer, t, positionOffset, rotationOffset, ignoreParentTransform).GetComponent<T>();
+            return GetPooledObject(objectPool, gameObject, 0, parent, Vector3.zero, Quaternion.identity, gameObject.transform.localScale, false);
         }
 
-        // Get a pooled object in a certain orientation (like a bullet being shot).
+        /// <summary>
+        /// Returns a component T from an object from the specified pool. Creates a new object and adds it to the pool if necessary.
+        /// Will throw an error if the component is not found.
+        /// </summary>
+        /// <typeparam name="T">A Unity component</typeparam>
+        /// <param name="objectPool">The active pool</param>
+        /// <param name="gameObject">The prefab to be used to create a new object</param>
+        /// <param name="parent">The transform under which a new object should be placed</param>
+        /// <returns>The component T of an active object from the requested pool</returns>
+        private static T GetPooledObject<T>(List<GameObject> objectPool, GameObject gameObject, Transform parent) where T : Component
+        {
+            return GetPooledObject(objectPool, gameObject, parent).GetComponent<T>();
+        }
```

Comment:
my IDE is saying this is unused, is it needed/should it be public?

Reply:
This should also be marked as obsolete now, shouldn't it? It's being called by the obsolete functions below.

Reply:
made it public